pkg/compact/downsample/downsample.go

```diff
@@ -598,21 +598,16 @@ type AggrChunk []byte
 // EncodeAggrChunk encodes a new aggregate chunk from the array of chunks for each aggregate.
 // Each array entry corresponds to the respective AggrType number.
 func EncodeAggrChunk(chks [5]chunkenc.Chunk) AggrChunk {
-	var mask byte
-	var all []chunkenc.Chunk
-
-	for i, c := range chks {
-		if c != nil {
-			mask |= 1 << (7 - uint(i))
-			all = append(all, c)
-		}
-	}
-
 	var b []byte
-	b = append(b, mask)
 	buf := [8]byte{}
 
-	for _, c := range all {
+	for _, c := range chks {
+		// Unset aggregates are marked with a zero length entry.
+		if c == nil {
+			n := binary.PutUvarint(buf[:], 0)
```

Comment:
Here just set 0 to `l` variable. No need for all of this and `continue`

Reply:
We need this as an indicator during deserialisation to know that the aggregate is explicitly unset.